### ISSUE TITLE
instalação

### ISSUE THREAD

**Issue by carlozousa** [scalargis/scalargis-server#19]
Estou com dificuldades em efectuar a instalação. No passo para o linux não consegui efectuar o comando 

python3 -m venv venv source venv/bin/activate

e o erro que me dá (unable to create directory)

diz-me que estou a lançar o python3 de forma errada.

Poderão dar uma ajuda?

**Comment by ricardogsena** [comment]
Boa tarde, Carlos

Antes de mais, o meu obrigado pelo interesse na utilização da plataforma Scalargis.

Essa instrução, que na verdade são dois comandos, permite instalar o ambiente virtual python para execução da aplicação. Deverá executar essa instrução  através de dois comandos independentes, pela seguinte ordem:

1 - python3 -m venv venv
2 - source venv/bin/activate

Esses comandos devem ser executados dentro da diretoria raiz do scalargis-server (para onde foi feito o git clone).

**Comment by carlozousa** [comment]
Boa tarde Ricardo,

Obrigado pela rápida resposta.
Com a execução dos restantes comandos, acabo por ser confrontado com uma dependência do flask que não sei se tem a ver com a minha versão do ubuntu, ou se há alguma exigência em termos de versão do linux/ubuntu que eu tenha de usar para experimentar o scalargis.
O erro é como segue:

<--
(venv) carlossousa@cartography:~/scalargis-server$ pip install -r requirements.t                           xt
ERROR: Could not find a version that satisfies the requirement Flask~=2.2.0 (fro                           m versions: 0.1, 0.2, 0.3, 0.3.1, 0.4, 0.5, 0.5.1, 0.5.2, 0.6, 0.6.1, 0.7, 0.7.1                           , 0.7.2, 0.8, 0.8.1, 0.9, 0.10, 0.10.1, 0.11, 0.11.1, 0.12, 0.12.1, 0.12.2, 0.12                           .3, 0.12.4, 0.12.5, 1.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.1.0, 1.1.1, 1.1.2, 1.1.3,                            1.1.4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.0.3)
ERROR: No matching distribution found for Flask~=2.2.0
(venv) carlossousa@cartography:~/scalargis-server$
-->

Obrigado novamente

**Comment by ricardogsena** [comment]
Esse que poderá ser alguma limitação provocada pela versão do python que está instalada nessa máquina. Com o ambiente virtual ativado (venv), pode verificar a versão através do seguinte comando:

python --version

Convém que a versão seja igual ou superior à 3.9.

**Comment by carlozousa** [comment]
Bom dia!

Tendo resolvido as dependências necessárias, sou confrontado com a necessidade de instalar adicionalmente o libgdal-dev.
As instruções seguintes estão em branco para a criação da base de dados e a instalação da extensão postgis através do "PGAdim"? Deduzo que seja o "PGAdmin".
Já fiz instalações manuais de Postgres/Postgis e penso que falta também, referência à estrutura da Base de Dados em si.
As referências à utilização do FLASK e respectivas confirguações, também estão em branco.

Não sei se isto é o local apropriado mas tendo em conta que vou ter de procurar outras fontes para complementar o que falta neste "how-to", depois partilharei convosco esse documento para que depois façam, ou não, um "how-to" mais completo. As dictomias do "it just works", apanagio do mundo "windows" e "macos" retiram toda e qualquer vontade de adoptar novas tecnologias e reutilizar código remetendo-o para a obscuridão.

Relembro que para a adopção e reutilização de código-fonte, é sempre deixar um rasto documental senão o projecto morre à nascença por falta de adeptos.

Um bom dia e boa semana de trabalho.

Obrigado!

**Comment by ricardogsena** [comment]
Concordo totalmente com as suas observações sobre a ausência de documentação. Essa é a principal razão de não ter sido feita uma divulgação pública do projeto, ainda há muito aspetos que necessitam de ser devidamente desenvolvidos. Esperamos resolver essas lacunas durante o presente ano, toda a ajuda será bem-vinda, pelo que agradeço desde já a sua disponibilidade de partilhar esse documento de apoio à instalação.

No que respeita à criação da base de dados, ela terá que ser criada ou existir previamente numa instância do PostgreSQL. Terá também que se especificar a string de ligação à base de dados no ficheiro[ scalargis/instance/config.py](https://github.com/scalargis/scalargis-server/blob/8b06b763256ab93e197b401e2ab92b3cc6d8a909/scalargis/instance/config.py#L12):

`SQLALCHEMY_DATABASE_URI = "postgresql+psycopg2://postgres:postgres@localhost:5432/scalargis"`

Caso a string de ligação à base de dados esteja correta e seja possível estabelecer essa ligação,  as tabelas de suporte à aplicação são criadas automaticamente quando se executa a aplicação e preenchidas com alguns dados de exemplo (visualizador de demonstração). Por defeito, as tabelas são criadas no schema _scalargis_.
